Makefile.generated_files

```diff
@@ -12,6 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Don't allow users to call this directly.  There are too many variables this
+# assumes to inherit from the main Makefile.  This is not a user-facing file.
+ifeq ($(CALLED_FROM_MAIN_MAKEFILE),)
+    $(error Please use the main Makefile to access these rules)
```

Comment:
I would add it to the comment then. It toke me a little time to find out where in the main Makefile this is referenced. I've searched for the filename "Makefile.generated_files" that is not used directly in the main Makefile. I would lgtm this anyway as it was an improvement even without this comment.